### PR TITLE
AArch64: disable interrupts in user_exc_leave.

### PR DESCRIPTION
**Pull request by pj1031999**
Disable interrupts in `user_exc_leave`.

### CODE REVIEW

File: sys/aarch64/genassym.cf [hunk 1]
```diff
@@ -5,6 +5,8 @@ include <sys/proc.h>
 include <aarch64/mcontext.h>
 include <aarch64/armreg.h>
 
+define DAIF_I_MASKED DAIF_I_MASKED
```
Comment (cahirwpz) [inline sys/aarch64/genassym.cf:8]: Please add `-Werror` to `ASFLAGS` in `build/flags.mk` and clean up warnings.

File: sys/aarch64/evec.S [hunk 2]
```diff
@@ -165,6 +165,9 @@ ENTRY(handle_user_trap)
         mov     x0, sp
         bl      user_trap_handler
 user_exc_leave:
+        /* disable interrupts */
+        msr     daifset, (DAIF_I_MASKED >> 6)
```
Comment (cahirwpz) [inline sys/aarch64/evec.S:169]: Well... this is confusing.

Some observations:
 * `msr daifset, #x` and `msr daifclr, #x` use `DAIF_I_MASKED` shifted right by `6`,
 * `mrs %r, daif` mirrors interrupt flags from current `PSR`,
 *  in our code `DAIF_I_MASKED` is the same as `PSR_I`.

Maybe we should clearly separate use of `PSR_*` and `DAIF_*` ? I would much prefer that `DAIF_I` (without `MASKED` suffix if possible) was used only with `daifclr` and `daifset` registers and `PSR_I` with `daif`. I'd add a comment in `armreg.h` about this oddity.

**Review by cahirwpz — COMMENTED** [review]
This is an important change but some cleanups would be welcome.

File: sys/aarch64/genassym.cf (hunk 1 above)
Comment (pj1031999) [reply sys/aarch64/genassym.cf]: done

File: sys/aarch64/evec.S (hunk 2 above)
Comment (pj1031999) [reply sys/aarch64/evec.S]: done

**Review by cahirwpz — APPROVED** [review]
LGTM

**Pull request merged**